components/user/UserProfile.js

```diff
@@ -68,34 +69,77 @@ function UserProfile({ BASE_URL, data }) {
         <div className="flex flex-wrap justify-center">
           {data.tags &&
             data.tags.map((tag) => (
-              <Tag name={tag} key={tag.toLowerCase()} onClick={() => router.push(`/search?keyword=${tag.toLowerCase()}`)} />
+              <Tag
+                name={tag}
+                key={tag.toLowerCase()}
+                onClick={() =>
+                  router.push(`/search?keyword=${tag.toLowerCase()}`)
+                }
+              />
             ))}
         </div>
       )}
 
       {/* Passed Ref object as the ref attribute to the JSX of the DOM node of QR */}
-      <div className="flex justify-center my-4" ref={qrRef}>
-        {qrShow && (
-          <QRCodeCanvas
-            className="border border-white"
-            value={`${BASE_URL}/${data.username}`}
-            size={fallbackImageSize * 2}
-          />
-        )}
-      </div>
-      <div className="flex justify-center mb-4">
-        {qrShow && (
-          <Button primary={true} onClick={downloadQR}>
-            Download QR code
-          </Button>
-        )}
-      </div>
+      <Modal show={qrShow} setShow={setQrShow} modalStyles="w-fit m-auto">
+        <div className="flex flex-col items-center justify-center px-8">
+          <div>
+            <div className="flex justify-center my-4" ref={qrRef}>
+              {qrShow && (
+                <QRCodeCanvas
+                  className="border border-white"
+                  value={`${BASE_URL}/${data.username}`}
+                  size={fallbackImageSize * 2}
+                />
+              )}
+            </div>
+            <div className="w-[100%] px-2 mx-auto flex justify-center mb-4">
+              {qrShow && (
+                <Button primary={true} onClick={downloadQR}>
+                  Download QR code
+                </Button>
+              )}
+            </div>
+          </div>
+          {qrShow && (
+            <>
+              <div className="h-full m-4 p-2 flex flex-row items-start justify-center space-x-2">
+                {socials.map(({ href, Icon, includeText }) => (
+                  <a
+                    key={href}
+                    href={`${href}${BASE_URL}/${data.username}${
+                      includeText
+                        ? `&text=${encodeURIComponent(
+                            `Check out ${data.name}'s profile on #LinkFree.io`
+                          )}`
+                        : ""
+                    }`}
+                    target="_blank"
+                    className="rounded-full p-2 border border-primary-low-medium hover:border-secondary-high hover:text-secondary-high dark:hover:text-primary-low dark:hover:border-primary-low-medium dark:hover:bg-primary-medium-low cursor-pointer  dark:bg-primary-medium"
+                  >
+                    <Icon size={24} />
+                  </a>
+                ))}
+              </div>
+              <div className="w-[100%] flex items-center justify-center">
```

Comment:
I think `w-[100%]` is the same as `w-full`? Which would be more consistent with what we use

Reply:
I looked into the code. w-full looks more consistent. I'll replace w-[100%] with it.

Reply:
@shoebilyas123 you marked this as resolved but I don't see any change?

Reply:
I have marked as unresolved

Reply:
```suggestion
              <div className="w-full flex items-center justify-center">
```